pubtools/pulplib/_impl/model/repository/base.py

```diff
@@ -471,8 +470,34 @@ def remove_content(self, **kwargs):
         # version of this method will support some "criteria".  Let's not fix the
         # argument order at least until then.
 
-        type_ids = kwargs.get("type_ids")
-        return f_proxy(self._client._do_unassociate(self.id, type_ids))
+        # start down the path of using Criteria per this issue:
+        # https://github.com/release-engineering/pubtools-pulplib/issues/62
+        # there seems to be pretty complex handling of Criteria
+        # for serialization in the search API, and it is unclear which parts
+        # might also be necessary to use for content removal.
+        # If any or all of the same handling is needed, it would be beneficial
+        # to encapsulate the preparation of a criteria JSON object in some
+        # (more generically named) functions or a class to avoid duplicating code.
+        # for reference see search_content, _impl.client.Client._search_repo_units,
+        # _impl.client.Client._search, and _impl.client.search.search_for_criteria
+        criteria = kwargs.get("criteria")
```

Comment:
Internally making use of a 'criteria' object seems like a good idea, but I'd rather we don't accept a criteria argument to this function yet.  Can we please drop this so the "if criteria is None" code path is the only path?

This is public API, if we support 'criteria' we need to at least document it, also document and test which types do/don't work.  I think it's premature to do that if we're not ready to support it "properly" yet.

Reply:
Good point, I have made the change to drop a `criteria` kwarg and instead stick with `type_ids`.